scipy/stats/tests/test_stats.py

```diff
@@ -2201,15 +2201,15 @@ def test_axes(self):
         data5 = [40, 30, 30, 30]
         arr = np.array([data1, data2, data3, data4, data5])
 
-        vals = stats.mode(arr, axis=None)
-        assert_equal(vals[0], np.array([30]))
-        assert_equal(vals[1], np.array([8]))
+        vals = stats.mode(arr, axis=None, keepdims=True)
+        assert_equal(vals[0], np.array([[30]]))
+        assert_equal(vals[1], np.array([[8]]))
```

Comment:
In other cases, I'll add `keepdims=True`.
Here, I also had to change the test because of the inconsistency noted in gh-17214.